File: sdk/java/src/test/java/com/lightswitch/domain/KeywordTest.java
package com.lightswitch.domain;

import static org.assertj.core.api.Assertions.*;

import java.util.List;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.Mock;

class KeywordTest {

	@Mock
	private Keyword keyword;

	private List<Property> properties;
	private String value;

	@BeforeEach
	void createKeyword(){
		properties = List.of(new Property("이름", "이승"), new Property("회사", "회사이름"));
		value = "true";
		keyword = new Keyword(properties, value);
	}


	@Test
	void getProperties() {
		assertThat(keyword).isNotNull();
		assertThat(keyword.getProperties().isEmpty()).isFalse();
		assertThat(properties.isEmpty()).isFalse();
		assertThat(keyword.getProperties()).isEqualTo(properties);
	}

	@Test
	void getValue() {
		assertThat(keyword).isNotNull();
		assertThat(keyword.getValue()).isEqualTo(value);
	}
}

File: backend/core-service/src/test/kotlin/com/lightswitch/core/domain/flag/repository/entity/KeywordTest.kt
package com.lightswitch.core.domain.flag.repository.entity

import com.lightswitch.core.domain.flag.common.enum.FlagType
import com.lightswitch.core.domain.flag.repository.FlagRepository
import com.lightswitch.core.domain.flag.repository.KeywordRepository
import com.lightswitch.core.domain.member.entity.Member
import com.lightswitch.core.domain.member.repository.MemberRepository
import com.lightswitch.core.domain.member.service.MemberService
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.transaction.annotation.Transactional

@Transactional
@SpringBootTest
class KeywordTest {

    @Autowired
    private lateinit var keywordRepository: KeywordRepository

    @Autowired
    private lateinit var flagRepository: FlagRepository

    @Autowired
    private lateinit var memberRepository: MemberRepository

    @Autowired
    private lateinit var memberService: MemberService

    var flag: Flag? = null

    @BeforeEach
    fun setUp() {
        memberRepository.findAllAByDeletedAtIsNull().map {
            memberService.deleteUser(it.memberId!!)
        }

        val savedMember = memberRepository.save(
            Member(
                lastName = "test",
                firstName = "test",
                telNumber = "01012345678",
                email = "test@gmail.com",
                password = "test",
            )
        )

        val savedFlag = flagRepository.save(
            Flag(
                title = "test",
                description = "test",
                type = FlagType.BOOLEAN,
                maintainer = savedMember
            )
        )
        flag = savedFlag
    }

    @Test
    fun `keyword 저장 테스트`() {
        // given
        val keyword1 = Keyword(
            flag = flag!!, description = "test", value = "test"
        )

        val keyword2 = Keyword(
            flag = flag!!, description = "test2", value = "test2"
        )

        // when
        val keywordList = keywordRepository.saveAll(listOf(keyword1, keyword2))

        // then
        assertThat(keywordList).hasSize(2)
        assertThat(keywordList.first().description).isEqualTo("test")
        assertThat(keywordList.first().value).isEqualTo("test")
        assertThat(keywordList.last().description).isEqualTo("test2")
        assertThat(keywordList.last().value).isEqualTo("test2")
    }

}